src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -499,6 +499,11 @@ public int available() throws IOException {
      * If cleanup of native resources is needed, other mechanisms such as
      * {@linkplain java.lang.ref.Cleaner} should be used.
      *
+     * @apiNote
+     * When this stream has an associated channel, this method may invoke
```

Comment:
I don't think a method can have more than one apiNote so the existing note will run into this text. Replacing @apiNote with a paragraph tag will help.

As regards the wording, then here's a suggestion that relies on the specification and avoids giving the impression that the close method directly calls itself recursively:

"If this stream has an associated channel then this method will close the channel, which in turn will close this stream. Subclasses that override this method should be prepared to handle possible reentrant invocation."

Reply:
Thanks - fixed in 90a130783.